src/ngraph/runtime/plaidml/plaidml_ops_group_convolution.cpp

```diff
@@ -0,0 +1,109 @@
+    Shape out_shape(src_shape.size());
+    for (std::size_t idx = 0; idx < src_shape.size(); ++idx)
+    {
+        out_shape[idx] = src_shape.at(m_output_axes.at(idx));
+    }
+    set_output_type(0, m_src->get_element_type(), out_shape);
+}
+
+std::shared_ptr<ngraph::Node>
+    ngraph::runtime::plaidml::op::GroupConvolution::copy_with_new_args(const NodeVector& new_args) const
+{
+    if (new_args.size() != 2)
+    {
+        throw ngraph_error{"PlaidMLGroupConvolution requires two inputs (data and filters)"};
+    }
+    return std::make_shared<GroupConvolution>(
+        m_src, new_args, m_data_axes, m_filters_axes, m_output_axes);
+}
+
+// GroupConvolution implements a grouped convolution, with optional striding, padding, and dilation.
+void ngraph::runtime::plaidml::ImplGroupConvolution::Apply()
+{
+
+    this->check_inputs(2);
+    this->check_outputs(1);
+
+    const auto& image = op_input(0);
+    const auto& filter = op_input(1);
+    
+    auto rank = op().get_input_shape(0).size() - 2;
+    const auto& groups = op().get_groups();
+    const auto& padding_above = op().get_padding_above();
+    const auto& padding_below = op().get_padding_below();
+    const auto& strides = op().get_window_movement_strides();
+    const auto& filter_dilation = op().get_window_dilation_strides();
+    const auto& data_dilation = op().get_data_dilation_strides();
+
+    const auto& grps = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(groups));
+    const auto& dd0 = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(data_dilation[0]));
+    const auto& dd1 = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(data_dilation[1]));
+    const auto& fd0 = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(filter_dilation[0]));
+    const auto& fd1 = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(filter_dilation[1])); 
+    const auto& pxb = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(padding_below[0]));
+    const auto& pyb = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(padding_below[1]));
+    const auto& pxa = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(padding_above[0]));
+    const auto& pya = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(padding_above[1]));
+    const auto& sx = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(strides[0]));
+    const auto& sy = static_cast<::vertexai::plaidml::variable>(static_cast<const int64_t>(strides[1]));
+
+    this->set_output(::vertexai::plaidml::function{R"(
+            function (I[N, CI, XI0, XI1], F[CO, FCI, XF0, XF1], DD0, DD1, FD0, FD1, G, PXB, PYB, PXA, PYA, SX, SY) -> (O) {
+                O[n, (CO/G) * g + co, x, y: N, CO, ((DD0 * (XI0 - 1) + PXA + PXB) - (FD0 * (XF0 - 1)) + SX) / SX, ((DD1 * (XI1 - 1) + PYA + PYB) - (FD1 * (XF1 - 1)) + SY) / SY] = 
```

Comment:
Did you test that the output dimensions were right under slight variations of the input dimensions? Especially in cases where strides, dilations, data_dilations, or a combination thereof are not 1, it can be really easy to get off-by-one errors in output dimensions in limited cases.

Not sure whether such tests should go in permanently, that might add up to a lot of very similar tests.

Reply:
I have a suite of tests that check for correctness of shapes/values for various parameter values of strides/window dilations/data dilations/padding. Those are available in `test/backend/fused_op.in.cpp`. Let me know if you meant something else other than the ones in there. Thanks!

Reply:
I meant a more thorough suite, in particular one testing a variety of input shapes against various fixed collections of stride/dilation parameters. The current tests are all using the same input data size, and it might happen to be one that works (the off-by-one errors I'm referring to are often cases like "if input_shape % strides == 1 then fail else succeed" which a single input size wouldn't necessarily catch)

Reply:
Also, it's really just the output dimensions I'm worried about for this, it seems unlikely that errors would crop up in the values in only these cases.

Reply:
Fair enough - I'll add those in.

Reply:
Added in a couple of new tests, and the op is still acting as expected.